### PR TITLE
document use-cases and criteria for design evaluation

### PR DESCRIPTION
**Pull request by lstreckeisen**

### CODE REVIEW

File: SA/design_concept/content/design/design_evaluation.typ [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#figure(
+    ```hs
+    addition :: Num a => a -> a -> a
+    addition a b = (+) a b
+    ```,
+    caption: "Addition function for design evaluation"
+)<design_eval_addition>
+
+== Even numbers from 1 to 10
+@design_eval_even has been chosen to evalute designs for list comprehension.
```
Comment (fl3pp) [inline SA/design_concept/content/design/design_evaluation.typ:16]:
```suggestion
@design_eval_even has been chosen to evalute designs for list handling.
```

File: SA/design_concept/content/design/functional_requirements.typ [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            The student user is the main user of VisualFP and therefore the main influence on the visualization design.
+            The user wants to learn functional programming using VisualFP.
+            They want to do that by visually composing functions in a simple UI.
+            The UI should simplify the understanding of functional concepts that many beginners have difficulties with.
+        ]
+    ),
+    terms.item(
+        "Expert User",
+        [
+            The expert user is an experienced professional that wants to use VisualFP to help them to better understand their code.
```
Comment (fl3pp) [inline SA/design_concept/content/design/functional_requirements.typ:20]:
```suggestion
            The expert user is an experienced professional that wants to use VisualFP to help them understand their code better.
```

File: SA/design_concept/content/design/functional_requirements.typ [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            For that, they want to be able to import their existing Haskell project into VisualFP.
```
Comment (fl3pp) [inline SA/design_concept/content/design/functional_requirements.typ:21]:
```suggestion
            For that, they want to import their existing Haskell project into VisualFP.
```

File: SA/design_concept/content/design/functional_requirements.typ [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+)<use_case_diagram>
+
+=== UC1 - Composition of simple function
+A user wants to compose a simple function using pre-defined functions and e.g. Integer parameters.
+
+=== UC2 - Function execution
+A user wants to execute their visually composed functions.
+
+=== UC3 - Composition of function using list comprehension
+A user wants to compose a function using list comprehension.
```
Comment (fl3pp) [inline SA/design_concept/content/design/functional_requirements.typ:43]: Not sure whether we should really focus on list comprehension. It is just syntactic sugar of Haskell, after all

File: SA/design_concept/content/design/functional_requirements.typ [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+=== UC4 - Composition of recursive function
+A user wants to compose a function that is defined by itself.
+
+=== UC5 - Composition of higher-order function
+A uswer wants to compose functions that take other functions as their argument.
+
+=== UC6 - Composition of custom data types
+A user wants to be able to create their own data types.
+
+=== UC7 - Saving of project file
```
Comment (fl3pp) [inline SA/design_concept/content/design/functional_requirements.typ:54]: Is a project file a source-file, a Cabal file or something of our own?

Comment (lstreckeisen) [reply SA/design_concept/content/design/functional_requirements.typ:54]: I intentionally left that open. Ideally that would be a file containing the Haskell AST, for the prototype something of our own. But I think that is something to clarify later in the document.

File: SA/design_concept/content/design/functional_requirements.typ (hunk 4 above)
Comment (lstreckeisen) [reply SA/design_concept/content/design/functional_requirements.typ:43]: I did think about visualizations for list comprehension for a few hours and it is indeed not the easiest feature to visualize. Especially when trying to support all possible cases.
So I'm inclined to drop it. Instead we could/should include the do notation as a use case.

File: SA/design_concept/content/design/functional_requirements.typ (hunk 5 above)
Comment (fl3pp) [reply SA/design_concept/content/design/functional_requirements.typ]: Okey, but in that case I would chose another nomenclature. I interpret _project file_ as something defining the project, such as `pom.xml` or `*.csproj` files

File: SA/design_concept/content/design/functional_requirements.typ (hunk 4 above)
Comment (fl3pp) [reply SA/design_concept/content/design/functional_requirements.typ:43]: We could think about the do notation, but I think we will have lots of thinks to do with even just the core language features, let alone the syntactic extensions.

File: SA/design_concept/content/design/functional_requirements.typ (hunk 5 above)
Comment (fl3pp) [reply SA/design_concept/content/design/functional_requirements.typ]: great, thanks!

File: SA/design_concept/content/design/functional_requirements.typ (hunk 4 above)
Comment (lstreckeisen) [reply SA/design_concept/content/design/functional_requirements.typ:43]: I'll rewrite the use-case to general list handling

**Review by fl3pp — APPROVED** [review]

**Pull request merged**